packages/amplify-category-api/src/graphql-transformer/transformer-factory.ts

```diff
@@ -102,7 +102,9 @@ const getTransformerFactoryV2 = (
   ];
 
   if (options?.addSearchableTransformer) {
-    transformerList.push(new SearchableModelTransformerV2());
+    const resourceDirParts = resourceDir.split(path.sep);
+    const apiName = resourceDirParts[resourceDirParts.length - 1];
+    transformerList.push(new SearchableModelTransformerV2(apiName));
```

Comment:
I'll leave it upto you to take a call, but instead of passing the `apiName` all the way down, we could fetch it wherever needed using a utility method that does something like:
```
const meta = stateManager.getMeta();
const resourceName = Object.entries(meta[AmplifyCategories.API]).find(
  ([, value]) => (value as { service: string }).service === AmplifySupportedService.APPSYNC,
)?.[0];
```

Reply:
I'll take a look at doing this. I'm seen some customer apps where there are multiple directories under API, but I'm guessing that what you're doing here by filtering to APPSYNC should mitigate that issue.

Reply:
Actually, I'd prefer to leave as-is, it's a little uglier, but since we've just smoke tested the behavior using the current impl, I'd rather leave it as close to the same as possible.